src/transaction/components/TransactionProvider.tsx

```diff
@@ -191,15 +195,19 @@ export function TransactionProvider({
         transactionReceipts: lifeCycleStatus.statusData.transactionReceipts,
       });
     }
-    // Emit Status
-    onStatus?.(lifeCycleStatus);
+    // Emit Status on status change
+    if (prevStatusEmitted !== lifeCycleStatus.statusName) {
+      onStatus?.(lifeCycleStatus);
+      setPrevStatusEmitted(lifeCycleStatus.statusName);
+    }
```

Comment:
are there any other use cases aside from subsequent transactions that we want to emit duplicate statuses for?

e.g. `init`?

I'd also add some comments here on why this is an edge case (multiple pending transactions)

Reply:
also nit but if we can make this something like

```
if (previousStatus !== currentStatus) {
 // some comment here on edge case
}
``